integration/https_test.go

```diff
@@ -61,7 +62,7 @@ func (s *HTTPSSuite) TestWithSNIConfigRoute(c *check.C) {
 	defer backend1.Close()
 	defer backend2.Close()
 
-	time.Sleep(2000 * time.Millisecond)
+	try.Sleep(2000 * time.Millisecond)
```

Comment:
IIUC then the sleep here are only necessary in order to wait for the test server to be up and running.

Why not replace this by `try.Do()`s that attempt to TCP-connect to each server?

Same for the other sleeps in this file.